extensions/amp-story-player/validator-amp-story-player.protoascii

```diff
@@ -28,9 +28,7 @@ tags: {  # <amp-story-player>
   html_format: AMP
   tag_name: "AMP-STORY-PLAYER"
   requires_extension: "amp-story-player"
-  child_tags: {
-    child_tag_name_oneof: "A"
-  }
+  descendant_tag_list: "amp-story-player-allowed-descendants"
```

Comment:
Just checking: You intend to switch from a child (ie: direct) to descendant (ie: entire subtree) constraint here, correct?

Reply:
Yep! That makes me think about something though: If we wanted to enforce the following tree structure:
```

<amp-story-player ...>
   <a>
      <span> ... </span>
   </a>
</amp-story-player>
```

We would need to add `mandatory_ancestor: "AMP-STORY-PLAYER" to the `<a>` tag in `validator-main.protoascii`, correct? But that would make it general for all `<a>` tags in all AMP documents? 🤔

Reply:
Yes, with the current rule system. In general, tree structure rules are one of the weaker areas of the validation engine today. This is mostly because we use a streaming (SAX) parser model, rather than a DOM model.